kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -871,6 +871,14 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val AUTHENTICATION_CUSTOM_HTTP_BEARER_HEADER: ConfigEntry[String] =
+    buildConf("kyuubi.authentication.custom.http.bearer.header")
+      .doc("Http header for http bearer authentication in ThriftHttp protocol.")
+      .version("1.10.0")
+      .serverOnly
+      .stringConf
+      .createWithDefault("BearerAuthentication")
```

Comment:
We do not need this, just use `Authorization`.

Reply:
It is the general auth header.

Reply:
```
Authorization: Basic base64(user:password)
Authorization: Bearer $BEARER_TOKEN
Authorization: Negotiate ...
```

Reply:
I got you, currently the http.header.Authorization can not be transferred to server end.

because https://github.com/apache/kyuubi/blob/0165d5e3a26cc8353a9f7135e8fa4a9c4ae99e78/kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiConnection.java#L446-L480

Reply:
so, the workaround:

1. use other header for bearer Autorization
2. or change the KyuubiConnection to support Bearer auth mode

cc @pan3793

Reply:
I prefer solution 2,

add `HttpBearerAuthInterceptor`, similar with org.apache.kyuubi.jdbc.hive.auth.HttpBasicAuthInterceptor

Reply:
I just go through the code, the BearerScheme is not supported in httpclient5.

So, for solution 1, we have to upgrade to httpclient5.

Reply:
https://github.com/apache/httpcomponents-client/pull/399

Bearer token auth is supported since 5.3

Reply:
Fixed it in https://github.com/apache/kyuubi/pull/6624/commits/d60f198edf54800d953aac55fac7782b119e8d4b 

You can set sessionConf var `bearerToken` for bearer auth.